pkg/cloudprovider/providers/gce/gce.go

```diff
@@ -942,6 +942,9 @@ func (gce *GCECloud) firewallObject(name, region, desc string, sourceRanges nets
 	if err != nil {
```

Comment:
the computeHostTags method is kind of weird. I'd expect user supplied input to be used, and if not supplied (eg: empyt list) to use some default, but the method has an early return that ignores the given list of hosts. 

Looks like there's only 1 call site for computeHostTags, can you move the check:
```
	if len(gce.nodeTags) > 0 {
		return gce.nodeTags, nil
	}
```
up into the caller, so it's clear that we're not actually using the supplied hosts if the tags are supplied?